src/ray/protobuf/node_manager.proto

```diff
@@ -350,6 +351,21 @@ message GetTaskFailureCauseReply {
   bool fail_task_immediately = 2;
 }
 
+message DrainRayletRequest {
+  // The reason why the node will be drained.
+  // This information will be used to
+  // decide whether the request is rejectable.
+  ray.rpc.autoscaler.DrainNodeReason reason = 1;
+  // The detailed drain reason message.
+  // Used for observability.
+  string reason_message = 2;
+}
+
+message DrainRayletReply {
+  // Whether the drain request is accepted.
```

Comment:
Can you add more details here? so that the caller can understand the behavior precisely. 

1. if the node is already dead, it is set to true
2. if the RPC is failed, the value must be ignored

Reply:
I added the comments to `autoscaler.proto`. The comments apply there not here: if the node is dead, we won't even send the DrainRaylet request.